accepted/00000-ansible-integration.md

```diff
@@ -0,0 +1,327 @@
+  * external host SSH key: (inventory label, local ssh key, host, port, remote inventory path, remote playbook path)
+  * AWX API (...)
+  * A system already registered in Uyuni.
+
+This tool would receive a JSON input with the necessary information to feed the corresponding plugin and will return a JSON output with collected information.
+
+Output example:
+
+```json
+{
+	"inventory-label": "my-ansible-control-node",
+        "hosts": [
+		"my-ansible-managed-system1.foo.bar",
+		"my-ansible-managed-system2.foo.bar",
+		"my-ansible-managed-system3.foo.bar"
+	],
+	"local_stored_inventory": "/var/lib/spacewalk/ansible/inventory-label/inventory.yaml",
+	"optional_local_stored_playbooks": [
+		"/var/lib/spacewalk/ansible/inventory-label/playbooks/playbook1.yaml",
+		"/var/lib/spacewalk/ansible/inventory-label/playbooks/playbook2.yaml",
+		"/var/lib/spacewalk/ansible/inventory-label/playbooks/playbook3.yaml"
+	]
+}
+```
+
+As you can see, besides of reporting the hosts from the Ansible inventory, the idea is that we also stored it locally in the Uyuni server in order to being able to reuse it later to target those hosts with "salt-ssh" directly from the Uyuni server.
+
+In short, we would gather, for instance:
+- Ansible inventory (hosts)
+- Ansible SSH keys referenced in the inventory
+- Playbooks under specified "remote playbook path" (to be displayed in the UI - readonly mode - not editable)
+
+This would be stored under `/var/lib/spacewalk/ansible/inventory-label/` path in the Uyuni server. At the time of processing the inventory file by "ansible-gatherer", it needs to be tailored to adapt the path of the SSH keys to the local path in the Uyuni server. As done in script from Ansible Integration PoC [here](https://github.com/meaksh/uyuni-hacks/blob/master/scripts/ansible/import_systems_from_ansible_controller.py).
+
+With this information, Java is able to proceed creating these systems in Uyuni, as "Foreign / ANSIBLE" system type.
+
+- If any of those systems is already existing with "ANSIBLE" system type, then nothing to do
+- If the system is already registered but not with "ANSIBLE" system type, then add "ANSIBLE" system type.
+
+Of course, we need some new tables in the database, at least something like:
+
+```
+"suseAnsibleController" (label, type, org_1)
+"suseAnsibleControllerConfig" (control_node_id, parameter, value)
+"suseAnsibleControllerSystem" (system_id, control_node_id)
+"suseAnsibleControllerPlaybook" (control_node_id, local_path, content?)
+```
+
+And also create the new "ANSIBLE" system type, which should be an addon-system type compatible with Salt, Foreign and, in case this doesn't need any extra effors, maybe also for a potential "Management/ANSIBLE" (traditional client) system type. Here is the [SQL script](https://github.com/meaksh/uyuni-hacks/blob/master/scripts/ansible/add_ansible_entitleme.sql) used for PoC demo. It's also needed to adapt the rhn databaste functions implemented in the DB.
```

Comment:
Try to remove "may" and "maybe". The RFC should be clear what is wanted and what not.
In this case I would remove "Management" completly from the RFC.